mlir/include/mlir/Interfaces/CallInterfaces.h

```diff
@@ -14,6 +14,7 @@
 #ifndef MLIR_INTERFACES_CALLINTERFACES_H
 #define MLIR_INTERFACES_CALLINTERFACES_H
 
+#include "mlir/IR/OpImplementation.h"
```

Comment:
OpImplementation.h pulls in a lot of extra junk, can you use forward declares here instead for whatever is needed?

Reply:
No, because of `OpAsmParser::Argument`  usage, I think it is not possible to forward declare nested types.

Either that class should be hoisted in a new OpAsmParserArgument class (a bit outside of the scope of this patch), or the helpers need to be in a different header (maybe that is why FunctionImplementation.h exists).

What do you think, is it worth splitting the header to avoid that include?

Reply:
I plan to not split the header. Will likely merge on Monday, please tell me if you think splitting the header is better.